### PR TITLE
citra-qt: fixes to per game settings

### LINKED ISSUE #6238: Launching game deselects post-processing shader
Using Citra Nightly 1825 launching a game de-selects any post-processing shader and has to be manually re-enabled after launching a game.

### PR DESCRIPTION
**Pull request by vitor-k**
Removes some unused functions and fixes some issues related to the per-game settings.

fixes #6238

fixes the settings that were written as QMetaTypes, like `@Variant(\0\0\0\x85\0\0)`, which would make them unreadable by humans.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6298)

### CODE REVIEW

File: src/citra_qt/configuration/config.cpp [hunk 1]
```diff
@@ -181,6 +181,15 @@ void Config::WriteBasicSetting(const Settings::Setting<std::string>& setting) {
     qt_config->setValue(name, QString::fromStdString(value));
 }
 
+// Explicit u16 definition: Qt would store it as QMetaType otherwise, which is not human-readable
+template <>
+void Config::WriteBasicSetting(const Settings::Setting<u16>& setting) {
+    const QString name = QString::fromStdString(setting.GetLabel());
+    const u16& value = setting.GetValue();
+    qt_config->setValue(name + QStringLiteral("/default"), value == setting.GetDefault());
+    qt_config->setValue(name, static_cast<unsigned int>(value));
```
Comment (GPUCode) [inline src/citra_qt/configuration/config.cpp:190]:
```suggestion
    qt_config->setValue(name, static_cast<u32>(value));
```

File: src/citra_qt/configuration/config.cpp [hunk 2]
```diff
@@ -223,6 +232,20 @@ void Config::WriteGlobalSetting(const Settings::SwitchableSetting<std::string>&
...
+template <>
+void Config::WriteGlobalSetting(const Settings::SwitchableSetting<u16, true>& setting) {
+    const QString name = QString::fromStdString(setting.GetLabel());
+    const u16& value = setting.GetValue(global);
+    if (!global) {
+        qt_config->setValue(name + QStringLiteral("/use_global"), setting.UsingGlobal());
+    }
+    if (global || !setting.UsingGlobal()) {
+        qt_config->setValue(name + QStringLiteral("/default"), value == setting.GetDefault());
+        qt_config->setValue(name, static_cast<unsigned int>(value));
```
Comment (GPUCode) [inline src/citra_qt/configuration/config.cpp:245]: ditto

File: src/common/settings.h [hunk 3]
```diff
@@ -201,6 +201,16 @@ class Setting {
         return default_value;
     }
 
+    /**
+     * Sets the default setting to the given value.
+     *
+     * @param val The desired value
+     */
+    virtual void SetDefault(const Type& val) {
```
Comment (GPUCode) [inline src/common/settings.h:209]: This doesn't need to be virtual

**Comment by vitor-k** [comment]
Addressed review comments and changed the post processing name setting to be split into one for the regular post processing and one for the anaglyph (as they already select from two different pools of options). With this change the selected shader will be remembered when switching from anaglyph and non-anaglyph 3d options.
Interlaced/reverse interlaced is left out, but the UI was reworked so that the selected shader can't be changed when using it.
This reflects the code behavior better (you can't use a post processing shader at the same time as the interlaced 3d options, it'd overwrite it).

**Pull request merged**